### ISSUE TITLE
Rate limit for individual endpoint

### ISSUE THREAD

**Issue by Mikey-ShenSu** [rayriffy/elysia-rate-limit#24]
**Is your feature request related to a problem? Please describe.**
Currently I am rate limiting the project with 10 requests per minute, but is it possible we can have rate limit for individual endpoints or group of endpoints? For instance, you may only be able to request SMS verification code once every 60 seconds, but for retrieving the data you can do max 10 requests every minute. 

**Describe the solution you'd like**
An argument that set the rate limit for individual endpoint without sharing the limit across the connection. 

**Describe alternatives you've considered**
I’ve thought that I may introduce the rate limit in each endpoint, but it adds extra code and complexity.

**Comment by rayriffy** [comment]
I think it might be possible, let me get into that. Related [discord/elysiajs#1197625891778482276](https://discord.com/channels/1044804142461362206/1167744218479996970/threads/1197625891778482276)

**Comment by rayriffy** [comment]
I've created version `3.2.0-beta.0` with feature that might matched your requirements, can you give it a try? More information at #27

**Comment by Mikey-ShenSu** [comment]
> I've created version `3.2.0-beta.0` with feature that might matched your requirements, can you give it a try? More information at #27. Basically, you set plugin scoping to local then separate SMS verification api route as separate `new Elysia()` as one instance
> 
> ```ts
> const smsVerificationRoutes = new Elysia()
>   .use(rateLimit())
>   // your sms routes here
> 
> const apiRoutes = new Elysia()
>   .use(rateLimit())
>   // your api routes here
> 
> const server = new Elysia()
>   .use(smsVerificationRoutes)
>   .use(apiRoutes)
>   .listen(3000)
> ```

```bash
[elysia-rate-limit] failed to determine client address (reason: .requestIP() returns null)
35 |                 }
36 |             }
37 |         });
38 |         app.onError(async ({ request }) => {
39 |             if (app.server === null)
40 |                 throw new Error('Elysia is not initialized yet. Please call .listen() first.');
                           ^
error: Elysia is not initialized yet. Please call .listen() first.
      at F:\ddxx\api\node_modules\elysia-rate-limit\dist\services\plugin.js:40:23
      at F:\ddxx\api\node_modules\elysia-rate-limit\dist\services\plugin.js:38:30
GET - http://localhost:3002/items?type=textbooks failed
```

I think this error is related to my code in the `index.ts`:
```typescript
async beforeHandle() {
        await connectToMongoDB();
        await connectToRedis();
      },
```

Is there any method, that I can make the rate limit wait until those 2 connections finish? What I want to achieve here, is to re-use connections across the endpoint, so client dont need to call the db everytime.

**Comment by rayriffy** [comment]
Maybe finishing connection to db before starting defining any elysia instances? You can use `global` to set variable globally to reuse them

**Comment by rayriffy** [comment]
i've updated to `3.2.0-beta.1` with requirement of server to be initialized removed. you should be able to get pass through that throw statement now.

**Comment by rayriffy** [comment]
You tell me 😅. If it works as expected, then I will push to mainstream.

**Comment by rayriffy** [comment]
@Mikey-ShenSu is this works as expected?

**Comment by Mikey-ShenSu** [comment]
> @Mikey-ShenSu is this works as expected?

This works for me fine on my end, I will make an issue ticket if I encounter anything. Thanks for your work tho!

**Comment by rayriffy** [comment]
Thanks released at `3.2.0` (Ref: #34)
